cloudfront/wellcomelibrary.org/edge-lambda/src/catalogueApi.ts

```diff
@@ -56,3 +56,25 @@ export async function* apiQuery(
     }
   }
 }
+
+export async function findWorkWithIdentifierValue(
+  identifierValue: string,
+  identifierType?: string
+): Promise<Work | undefined> {
+  const resultList = apiQuery(identifierValue);
+
+  for await (const result of resultList) {
+    if (result.identifiers) {
+      const identifiers: Identifier[] = result.identifiers;
+      const hasMatchingId = identifiers.some(
+        (identifier) =>
+          identifier.value.toLowerCase() === identifierValue.toLowerCase() &&
+          (!identifierType || identifier.identifierType.id === identifierType)
```

Comment:
Since we're in an edge lambda please forgive this perf-optimisation bicycle storage solution: swapping the order of these 2 predicates (the value check and the type check) is marginally preferable because the value check is more computationally expensive

Reply:
Any benchmarks?

Reply:
There we have it
https://jsben.ch/bofnA